crates/bevy_render/src/render_graph/nodes/texture_node.rs

```diff
@@ -25,17 +27,57 @@ impl TextureNode {
             texture_descriptor,
             sampler_descriptor,
             handle,
+            has_changed: true,
         }
     }
 }
 
+impl TextureNode {
+    pub fn texture_descriptor(&self) -> &TextureDescriptor {
+        &self.texture_descriptor
+    }
+
+    pub fn texture_descriptor_mut(&mut self) -> &mut TextureDescriptor {
+        self.set_changed();
+        &mut self.texture_descriptor
+    }
+
+    pub fn sampler_descriptor(&self) -> &Option<SamplerDescriptor> {
+        &self.sampler_descriptor
+    }
+
+    pub fn sampler_descriptor_mut(&mut self) -> &mut Option<SamplerDescriptor> {
+        self.set_changed();
+        &mut self.sampler_descriptor
+    }
+
+    pub fn set_changed(&mut self) {
+        self.has_changed = true;
+    }
+}
+
 impl Node for TextureNode {
     fn output(&self) -> &[ResourceSlotInfo] {
-        static OUTPUT: &[ResourceSlotInfo] = &[ResourceSlotInfo {
-            name: Cow::Borrowed(TextureNode::TEXTURE),
+        static WITHOUT_SAMPLER: &[ResourceSlotInfo] = &[ResourceSlotInfo {
+            name: Cow::Borrowed(TextureNode::OUT_TEXTURE),
             resource_type: RenderResourceType::Texture,
         }];
-        OUTPUT
+        static WITH_SAMPLER: &[ResourceSlotInfo] = &[
+            ResourceSlotInfo {
+                name: Cow::Borrowed(TextureNode::OUT_TEXTURE),
+                resource_type: RenderResourceType::Texture,
+            },
+            ResourceSlotInfo {
+                name: Cow::Borrowed(TextureNode::OUT_SAMPLER),
+                resource_type: RenderResourceType::Sampler,
+            },
+        ];
+
+        if self.sampler_descriptor.is_none() {
```

Comment:
I would strongly prefer `Option::map_or` instead of an `if/else` branch
```rust
self.sampler_descriptor
    .map_or(WITHOUT_SAMPLER, |_| WITH_SAMPLER)
```